rcpchgrowth.js

```diff
@@ -80,161 +125,84 @@ function UK_WHO_SDS_CENTILE(
- * @customfunction
- */
-
-function UK_WHO_Centile(
-  birth_date,
-  observation_date,
-  gestation_weeks,
-  gestation_days,
-  sex,
-  measurement_method,
-  observation_value
-) {
-  var formData = {
-    birth_date: birth_date,
-    observation_date: observation_date,
-    sex: sex,
-    gestation_weeks: gestation_weeks,
-    gestation_days: gestation_days,
-    measurement_method: measurement_method,
-    observation_value: observation_value,
-  };
-  var jsonified = JSON.stringify(formData);
-
-  var request = {
-    url: url,
-    payload: jsonified,
-    method: "POST",
-    headers: {
-      "Content-Type": "application/json",
-    },
-  };
-
-  var url = "https://api.rcpch.ac.uk/uk-who/calculation";
-
-  var response = UrlFetchApp.fetch(url, request);
-
-  var results = JSON.parse(response);
-  var chronological_age = results.measurement_dates.chronological_decimal_age;
-  var corrected_age = results.measurement_dates.corrected_decimal_age;
-  var sds = results.measurement_calculated_values.sds;
-  var centile = results.measurement_calculated_values.centile;
+  let return_data;
+
+  switch (data_to_return) {
+    case "both":
+      return_data = [
+        [
+          corrected_sds,
+          chronological_sds,
+          corrected_centile,
+          chronological_centile,
+        ],
+      ];
+      break;
+    case "sds":
+      return_data = [[corrected_sds, chronological_sds]];
+      break;
+    case "centiles":
+      return_data = [[corrected_centile, chronological_centile]];
+      break;
+  }
 
-  return centile;
+  return return_data;
 }
 
 /**
- * Returns the chronological age as a decimal from 2 dates.
- * @param {date} birth_date. (must be YYY-MM-DD)
- * @param {date} observation_date. (must be YYY-MM-DD)
- * @param {integer} gestation weeks.
- * @param {integer} gestation days.
- * @param {string} sex: must be one of male or female
- * @param {string} measurement_method: must be one of height, weight, ofc or bmi
- * @param {number} observation_value:
- * @return The chronological age.
+ * Returns the chronological age as a decimal from 2 dates, and/or decimal age corrected for gestational age if premature (< 37 weeks gestation).
+ * @param {date} birth_date Must be date format
+ * @param {date} observation_date Must be date format
+ * @param {integer} gestation_weeks Must be integer
+ * @param {integer} gestation_days Must be integer
+ * @param {string} sex Must be one of male, female
+ * @param {string} measurement_method Must be one of height, weight, ofc, bmi
+ * @param {number} observation_value
+ * @param {string} primary_api_key Your primary_api_key
+ * @param {string} data_to_return OPTIONAL, default "both", must be one of "both", "chron", "corr"
+ * @return The requested decimal ages. If "both", returns in the following order: chronological age -> corrected age.
  * @customfunction
  */
-
-function chronological_age(
+function UK_WHO_DECIMAL_AGE(
```

Comment:
@anchit-chandran I would possibly call the function UK_WHO_CORRECTED_DECIMAL_AGE so it's explicit that it does GAC. Obvs for term babies it doesn't correct but it's still more explicit that way.